vhdbuilder/scripts/automate_version_bump.sh

```diff
@@ -47,10 +49,47 @@ update_image_version() {
     sed -i "s/${current_image_version}/${new_image_version}/g" pkg/agent/datamodel/sig_config_test.go
 }
 
-find_current_image_version "pkg/agent/datamodel/osimageconfig.go"
-set_git_config
-create_branch $branch_name
-update_image_version
+create_image_bump_pr() {
+    create_branch $branch_name
+    update_image_version
 
-set +x
-create_pull_request $new_image_version $github_access_token $branch_name $pr_title
\ No newline at end of file
+    set +x
+    create_pull_request $new_image_version $github_access_token $branch_name $pr_title
+    set -x
+}
+
+# This function cuts the official branch based off the commit ID that the builds were triggered from and tags it
+cut_official_branch() {
+    official_branch_name="official/v${new_image_version//.}"
+    official_tag="v0.${new_image_version//.}.0"
+    final_commit_hash=""
+    for build_id in $build_ids; do
```

Comment:
would it be cleaner if we move this section (from l66:76) to a separate function like `check_builds_same_commit`? That would make that validation logic separate from the script to cut the branch.

Reply:
I did not move it to a separate function because the check isn't too long/complicated. Can do that in a later PR though, will add a TODO if that is ok. @ganeshkumar5699

Reply:
yes, that sounds good :)